api/src/ddoc-extraction.js

```diff
@@ -59,6 +56,9 @@ const isUpdated = (settings, newDdoc) => {
         oldDdoc._attachments = newDdoc._attachments;
       }
 
+      // delete the obsolete app_settings so the docs will be comparable
+      delete oldDdoc.app_settings;
```

Comment:
would it be crazy to do this in a migration and then just presume we're fine at this stage?

Reply:
ddoc extraction runs before migrations on startup, and then after migrations on change, so we can't assume it's fine at this stage. However I think if it's implemented as a migration we don't need to do it here.